### PR TITLE
feat: add commit hash to comments

### LINKED ISSUE #663: Bot Deploys Comment Should Include Commit Hashes
I was trying to understand which commit corresponds to which deployment but the commit hashes don't seem to match based on what I see inside of the rendered dApp. 

The deployment links from the bot also do not include the commit hashes. 

Verifying by timestamp seems unnecessarily imprecise.

> [Deployment: Mon May 29 2023 10:18:51 (UTC)](https://64747bfbcc5035619c6b2463--ubiquity-dollar.netlify.app)
> [Deployment: Mon May 29 2023 11:48:24 (UTC)](https://647490fde519217706996dfd--ubiquity-dollar.netlify.app)
> [Deployment: Mon May 29 2023 13:27:19 (UTC)](https://6474a82b9837770a9c41cb82--ubiquity-dollar.netlify.app)
> [Deployment: Mon May 29 2023 13:38:49 (UTC)](https://6474aade82d509062db6fbe1--ubiquity-dollar.netlify.app)
> [Deployment: Mon May 29 2023 13:42:03 (UTC)](https://6474ab9fae27a57ec36fe779--ubiquity-dollar.netlify.app)

_Originally posted by @ubiquibot[bot] in https://github.com/ubiquity/ubiquity-dollar/issues/661#issuecomment-1566917449_

### PR DESCRIPTION
**Pull request by PhantomCracker**
Resolves #663

### CODE REVIEW

**Comment by 0x4007** [comment]
Can you open a pull request on your fork and link it here to prove that it works? Otherwise can you post any other proof that it works?

**Comment by PhantomCracker** [comment]
> Can you open a pull request on your fork and link it here to prove that it works? Otherwise can you post any other proof that it works?

Please check this PR: https://github.com/PhantomCracker/ubiquity-dollar/pull/7

**Comment by 0x4007** [comment]
I don't see the commit hashes there. What are you showing with your linked pull request? I'm not asking to see the code. I'm asking to see proof that it works according to the spec.

**Comment by PhantomCracker** [comment]
@pavlovcik recheck the PR please; here is also the proof of testing:
Screenshot 2023-06-18 at 18 02 06

**Comment by 0x4007** [comment]
I don't see any comments posted by the bot with the commit hashes on your fork.

**Comment by PhantomCracker** [comment]
Ok, I think that everything is ok now; i have a issue on my side with that Netlify, but i think that you can generate some logs to see if everything is ok here.
Thank you.

**Comment by rndquu** [comment]
As far as I understand this issue required to convert netlify deploy URL from `Deployment: Mon May 29 2023 10:18:51 (UTC)` to `Deployment: Mon May 29 2023 10:18:51 (UTC) | Commit: c123456`

@PhantomCracker could you show how it works **in your forked environment**?

**Comment by PhantomCracker** [comment]
@rndquu can you help with how can I do a deployment message on my fork like this? 
Screenshot 2023-06-22 at 10 43 17

**Comment by 0x4007** [comment]
> As far as I understand this issue required to convert netlify deploy URL from `Deployment: Mon May 29 2023 10:18:51 (UTC)` to `Deployment: Mon May 29 2023 10:18:51 (UTC) | Commit: c123456`
> 
> 
> 
> @PhantomCracker could you show how it works **in your forked environment**?

I think only the seven character commit hash is sufficient. Time stamps can be derived from the comment edit history times. But time stamps seem not particularly useful in my experience.

**Comment by rndquu** [comment]
> @rndquu can you help with how can I do a deployment message on my fork like this? Screenshot 2023-06-22 at 10 43 17

I guess you should add netlify app to your forked environment and use your own github PAT [here](https://github.com/ubiquity/ubiquity-dollar/blob/8de9a1cb98bb3597974beec9fe0ee9fe5ef2fa3a/.github/workflows/continuous-deploy.yml#L89) for testing purposes

**Comment by PhantomCracker** [comment]
I always receive this error on my fork:
Screenshot 2023-06-22 at 13 02 44
Maybe we can wait for another deployment message?

**Comment by rndquu** [comment]
> I always receive this error on my fork: Screenshot 2023-06-22 at 13 02 44 Maybe we can wait for another deployment message?

I guess [here](https://github.com/ubiquity/ubiquity-dollar/blob/8de9a1cb98bb3597974beec9fe0ee9fe5ef2fa3a/.github/workflows/continuous-deploy.yml#L43) you need to setup the `NETLIFY_SITE_ID_DEVELOPMENT` env variable in your github secrets

**Comment by PhantomCracker** [comment]
@rndquu i come back with another question, what access does cloudflare need for the CLOUDFLARE_API_TOKEN?

**Comment by rndquu** [comment]
> @rndquu i come back with another question, what access does cloudflare need for the CLOUDFLARE_API_TOKEN?

There is a template called `Edit Cloudflare Workers `. You could use this one on API token generation.

**Comment by PhantomCracker** [comment]
@rndquu  can you also please provide these two secrets?
secrets.UBIQUITY_BOUNTY_BOT_APP_ID
secrets.UBIQUITY_BOUNTY_BOT_PRIVATE_KEY

**Comment by rndquu** [comment]
> @rndquu can you also please provide these two secrets? secrets.UBIQUITY_BOUNTY_BOT_APP_ID secrets.UBIQUITY_BOUNTY_BOT_PRIVATE_KEY

You should setup you own bot's instance and set your own env values for `UBIQUITY_BOUNTY_BOT_APP_ID` and `UBIQUITY_BOUNTY_BOT_PRIVATE_KEY`

**Comment by PhantomCracker** [comment]
Please check the following GH action results: https://github.com/PhantomCracker/ubiquity-dollar/actions/runs/5445801433/jobs/9905580034
Here is also a screenshot:
Screenshot 2023-07-03 at 17 46 14

**Comment by rndquu** [comment]
@PhantomCracker pls check this PR https://github.com/rndquu/ubiquity-dollar/pull/20

It prints `[Deployment: Tue Jul 04 2023 08:22:14 (UTC) | Commit: 2c143d6c0018b081b8b8c865e6d9ff0647763c4a](https://a630927c.ubiquity-dollar-1cd.pages.dev/)` but the latest commit hash is `a82e8ca6894d37b50a3a336352362afb49e15101`. So they are expected to match.

**Comment by PhantomCracker** [comment]
> @PhantomCracker pls check this PR [rndquu#20](https://github.com/rndquu/ubiquity-dollar/pull/20)
> 
> It prints `[Deployment: Tue Jul 04 2023 08:22:14 (UTC) | Commit: 2c143d6c0018b081b8b8c865e6d9ff0647763c4a](https://a630927c.ubiquity-dollar-1cd.pages.dev/)` but the latest commit hash is `a82e8ca6894d37b50a3a336352362afb49e15101`. So they are expected to match.

Thank you for feedback, work in progress!

**Comment by PhantomCracker** [comment]
Hm... @rndquu i come back with different tests:
https://github.com/PhantomCracker/ubiquity-dollar/commit/dcd6a09fd116d210b173ca8569b94c8a1ab40a83#comments
and https://github.com/PhantomCracker/ubiquity-dollar/commit/6d39638c6e1a746a1db66b1df97133f01a497208#comments

I have also tried to create a new PR to test this: https://github.com/PhantomCracker/ubiquity-dollar/pull/8
As you can see, the SHA is:  a3b4d077cb2000fa5862d382856cb514c4054812 which is the same as in this action: https://github.com/PhantomCracker/ubiquity-dollar/actions/runs/5453519281/jobs/9922468500
But the last commit have the following SHA: 975a32998e1f9fee285bc2a038090ffe06d6ea05 which corresponds to this action: https://github.com/PhantomCracker/ubiquity-dollar/actions/runs/5453525015/jobs/9922481199
The difference between these two is that on the first action, we display the SHA for the PR, while on the second we display the SHA for the commit https://github.com/PhantomCracker/ubiquity-dollar/commit/975a32998e1f9fee285bc2a038090ffe06d6ea05#comments

Is this the expected functionality?

**Comment by rndquu** [comment]
> Hm... @rndquu i come back with different tests: [PhantomCracker@dcd6a09#comments](https://github.com/PhantomCracker/ubiquity-dollar/commit/dcd6a09fd116d210b173ca8569b94c8a1ab40a83#comments) and [PhantomCracker@6d39638#comments](https://github.com/PhantomCracker/ubiquity-dollar/commit/6d39638c6e1a746a1db66b1df97133f01a497208#comments)
> 
> I have also tried to create a new PR to test this: [PhantomCracker#8](https://github.com/PhantomCracker/ubiquity-dollar/pull/8) As you can see, the SHA is: [a3b4d07](https://github.com/ubiquity/ubiquity-dollar/commit/a3b4d077cb2000fa5862d382856cb514c4054812) which is the same as in this action: https://github.com/PhantomCracker/ubiquity-dollar/actions/runs/5453519281/jobs/9922468500 But the last commit have the following SHA: [975a329](https://github.com/ubiquity/ubiquity-dollar/commit/975a32998e1f9fee285bc2a038090ffe06d6ea05) which corresponds to this action: https://github.com/PhantomCracker/ubiquity-dollar/actions/runs/5453525015/jobs/9922481199 The difference between these two is that on the first action, we display the SHA for the PR, while on the second we display the SHA for the commit [PhantomCracker@975a329#comments](https://github.com/PhantomCracker/ubiquity-dollar/commit/975a32998e1f9fee285bc2a038090ffe06d6ea05#comments)
> 
> Is this the expected functionality?

pls merge the latest `development` branch and try again, commit hashes should be the same now

**Comment by PhantomCracker** [comment]
Indeed it works! 👍 🥳 
Proof of testing: https://github.com/PhantomCracker/ubiquity-dollar/pull/9

**Review by rndquu — APPROVED** [review]

File: utils/print-deployments-logs.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = async ({ github, context, fs }) => {
   const botCommentsArray = [];
 
   if (uniqueDeployUrl) {
-    defaultBody = `[Deployment: ${new Date()}](${uniqueDeployUrl})`;
+    defaultBody = `[Deployment: ${new Date()} | Commit: ${commitSha}](${uniqueDeployUrl})`;
```
Comment (0x4007) [inline utils/print-deployments-logs.js]:
```suggestion
    defaultBody = `- [${commitSha}](${uniqueDeployUrl})`;
```

Can you also calculate the seven letter commit SHA like how GitHub displays it? I'm unsure if they slice the end or if they calculate it differently but it should match what shows in the UI (which if I recall is seven characters- sorry I'm on my phone GitHub client now it's not easy for me to check)

**Review by 0x4007 — CHANGES_REQUESTED** [review]

File: utils/print-deployments-logs.js (hunk 1 above)
Comment (PhantomCracker) [reply utils/print-deployments-logs.js]: Sure, I will update it

**Comment by 0x4007** [comment]
I hope you verified that the way you're deriving the seven character SHA matches what GitHub displays or ill have to revert this.

**Comment by PhantomCracker** [comment]
> I hope you verified that the way you're deriving the seven character SHA matches what GitHub displays or ill have to revert this.

For "dcd6a09fd116d210b173ca8569b94c8a1ab40a83" we will display dcd6a09.
Please let me know if something is wrong and I will fix it 😄

**Pull request merged**